spec/tasks/adoptapet_sync_spec.rb

```diff
@@ -9,15 +9,15 @@
     let(:foster) { FactoryBot.create(:foster, region: "VA") }
     let!(:dog) { FactoryBot.create(:dog, status: 'adoptable', foster_id: foster.id, photos_attributes: [photo0,photo1,photo2,photo3,photo4].map(&:attributes)) }
     let!(:rake){ task.execute }
-    let(:recent_photos){ Dog.first.photos.sort_by{|photo| photo.updated_at}.reverse[0..3]}
+    let(:recent_photos){ Dog.first.photos.visible.sort_by{|photo| photo.position}}
     let(:photo_urls){ recent_photos.map{|photo| photo.photo.url(:large)} }
     let(:csv) { CSV.read( '/tmp/adoptapet/pets_VA.csv' ) }
 
     it "should include most recent four photos for a dog" do
       csv_photos = csv.first[16..19]
       # just confirming recent_photos are the ones we want to upload
       # expect most recent first
-      expect(recent_photos.map(&:updated_at)).to be_sorted(:descending)
+      expect(recent_photos.map(&:updated_at)).to be_sorted(:ascending)
```

Comment:
ascending means that earlier values precede later ones e.g. 2018/1/1 will sort ahead of 2018/1/2 so you're not going to get `should include most recent four photos for a dog` you're going to get `should include oldest four photos for a dog with the oldest one coming first`

Not sure if that is your exact problem or not but your `it/do` does not match your code

Reply:
Hey Matt.

Apologies I need to rewrite the spec description still.


The app code was changed so that instead of sorting for most recent photos, it should instead look at the `position` which is just an integer that sets the order of the photos.  We just want the photos in positions 1 thru 4.  

App code works right, just can’t seem to get the test right.

Reply:
No worries. Just trying to help ;)

Reply:
Should touch sort by position instead of updated_at?

Reply:
![Facepalm](https://media.giphy.com/media/vwI4mYEHP8k0w/giphy.gif)

Thanks!  Looks like it was that and a missing `[0..3]` to trim down the photo set.